contexts/v2/index.json

```diff
@@ -0,0 +1,80 @@
+{
+  "@context": {
+    "id": "@id",
+    "type": "@type",
+    "JsonWebKey2020": {
+      "@id": "https://w3id.org/security#JsonWebKey2020",
+      "@context": {
+        "controller": {
+          "@id": "https://w3id.org/security#controller"
+        },
+        "publicKeyJwk": {
+          "@id": "https://w3id.org/security#publicKeyJwk",
+          "@context": {
+            "@vocab": "https://www.iana.org/assignments/jose/jose.xhtml#"
+          }
```

Comment:
-1 to using @vocab in a security-related context. This seems dangerous, as the statement means that anything that is defined in `publicKeyJwk` will be assigned to the IANA registry including terms like `foo`, and `bar`, which clearly do not exist in the IANA registry.

If we are going to map semantics to IANA, we should make sure those terms exist in the IANA registry. The other option here is to just specify that the field is `@json`, which is what the IETF specs suggest.

My suggestion here is to explicitly use the fields declared by the JWK specification instead of just mapping anything that shows up in publicKeyJwk to the IANA registry. It opens up an extra attack surface that we shouldn't be opening up unless there is a very good reason to open it up.

Reply:
> This seems dangerous, as the statement means that anything that is defined in `publicKeyJwk` will be assigned to the IANA registry including terms like foo, and bar, which clearly do not exist in the IANA registry.

The question is... do we know what will be in the IANA registry 30 years from now?

Where will people be registering new JWK terms, if not with IANA?

Which authority maintains the terms registered to support interoperability for JWK?

Relevant sections of the JSON-LD 1.1 spec:

- https://www.w3.org/TR/json-ld11/#default-vocabulary
- https://www.w3.org/TR/json-ld11/#iana-security


> Given that JSON-LD allows the substitution of long IRIs with short terms, JSON-LD documents may expand considerably when processed and, in the worst case, the resulting data might consume all of the recipient's resources. Applications should treat any data with due skepticism.

sounds like this was already considered.

> As JSON-LD places no limits on the IRI schemes that may be used, and vocabulary-relative IRIs use string concatenation rather than IRI resolution, it is possible to construct IRIs that may be used maliciously, if dereferenced.

I think the better solution here, is to have a normative requirement in the data integrity spec, then all data integrity suites will be aware of how to comply with W3C requirements associated with vocabularies.